### PR TITLE
Corelight

### PR DESCRIPTION
**Pull request by averbn**
## Proposed Changes

  - Corelight Data Connector
  -
  -

### CODE REVIEW

**Comment by preetikr** [comment]
@shainw - Please review the parser. Thanks.

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
+{
+    "id": "Corelight",
+    "title": "Corelight",
+    "publisher": "Corelight",
+    "descriptionMarkdown": "The [Corelight](https://corelight.com/) data connector provides the capability to ingest [Corelight Zeek/Bro events](https://www3.corelight.com/zeek-logs-v3.0) into Azure Sentinel. Refer to [Corelight Logs documentation](https://corelight.com/about-zeek/how-zeek-works) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **Corelight** in queries and workbooks. [Follow steps to get this Kusto Function>](https://aka.ms/sentinel-Corelight-parser)",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "Corelight Events",
```
Comment (v-admahe) [inline DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:10]: Legend name should be datatype name. Could you please update it  to "Corelight_CL".

Thanks,
Aditi Maheshwari

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "Corelight\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1
```
Comment (v-admahe) [inline DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:35]: IsPreview Tag is missing

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json (hunk 2 above)
Comment (v-admahe) [inline DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:35]: Status should be 2. Please refer sample template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_template.json

Thanks,
Aditi Maheshwari

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    },
+    "permissions": {
+        "resourceProvider": [
```
Comment (v-admahe) [inline DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:38]: Permission is not appropriate .Please refer the sample template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_template.json

or you can use this text as well:


 "permissions": {
        "resourceProvider": [
            {
                "provider": "Microsoft.OperationalInsights/workspaces",
                "permissionsDisplayText": "read and write permissions are required.",
                "providerDisplayName": "Workspace",
                "scope": "Workspace",
                "requiredPermissions": {
                    "write": true,
                    "read": true,
                    "delete": true
                }
            },
            {
                "provider": "Microsoft.OperationalInsights/workspaces/sharedKeys",
                "permissionsDisplayText": "read permissions to shared keys for the workspace are required. [See the documentation to learn more about workspace keys](https://docs.microsoft.com/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key).",
                "providerDisplayName": "Keys",
                "scope": "Workspace",
                "requiredPermissions": {
                    "action": true
                }
            }
        ]

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@averbn  Could you please provide sample data (Corelight_CL) .Please include file in PR.

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
@averbn  Please resolve merge conflict.

Thanks,
Aditi Maheshwari

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json (hunk 1 above)
Comment (averbn) [reply DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:10]: done

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json (hunk 2 above)
Comment (averbn) [reply DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:35]: done

Comment (averbn) [reply DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:35]: done

File: DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json (hunk 3 above)
Comment (averbn) [reply DataConnectors/Corelight/Connector_LogAnalytics_agent_Corelight.json:38]: done

**Comment by v-jayakal** [comment]
@averbn , Sample data is needed as json file with an array of elements, Can you please refer to sample data available at https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Custom folder. One such an example file for reference is https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/Custom/BetterMTDAppLog_CL.json.
Please provide the data in the format that can be used for queries verification in connector and workbooks.
Please let us know if you need any information.

Thanks.

**Comment by averbn** [comment]
> @averbn , Sample data is needed as json file with an array of elements, Can you please refer to sample data available at https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Custom folder. One such an example file for reference is https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/Custom/BetterMTDAppLog_CL.json.
> Please provide the data in the format that can be used for queries verification in connector and workbooks.
> Please let us know if you need any information.
> 
> Thanks.

@v-jayakal, The example of each log has three fields: message, log_file, hostname(The  LA agent reads raw files from the directory). The main information of the log placed in the "Message" field and is parsed by the function on "fly" in the Sentinel. I`ll update the sample file.

**Review by v-admahe — DISMISSED** [review]

**Comment by v-admahe** [comment]
Please update the parse link in workbook to "https://aka.ms/sentinel-Corelight-parser"

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
@averbn There is typo mistake in parser alias name it should be Corelight .

Thanks,
Aditi Maheshwari

File: Parsers/Corelight/Corelight [hunk 4]
```diff
@@ -0,0 +1,621 @@
+// Usage Instruction :
```
Comment (shainw) [inline Parsers/Corelight/Corelight:1]: Generally we prefer these to have a .TXT extension for ease of offline usage.

File: Parsers/Corelight/Corelight [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. Corelight | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+let Corelight_view = view () {
+    Corelight_CL | where isnotempty(Message)
+| extend tmp = parse_json(Message)
+| evaluate bag_unpack(tmp)| extend path_parts = parse_path(log_file_s)
+    | extend EventType = extract("(^.*?)_\\d+", 1, tostring(path_parts["Filename"])),
+        EventVendor="Corelight",
+        EventProduct="Corelight Sensor",
+        SrcDvcHostname=column_ifexists('hostname_s', ''),
```
Comment (shainw) [inline Parsers/Corelight/Corelight:12]: sample data set has this field named 'hostname' not 'hostname_s'

File: Parsers/Corelight/Corelight [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+        EventEndTime=column_ifexists('ts', ''),
+        SrcDvcFile=column_ifexists('log_file_s', '')
```
Comment (shainw) [inline Parsers/Corelight/Corelight:14]: sample data set has this field named 'log_file' not 'log_file_s'

**Review by shainw — CHANGES_REQUESTED** [review]

File: Parsers/Corelight/Corelight (hunk 5 above)
Comment (averbn) [reply Parsers/Corelight/Corelight:12]: _s suffix will be added automatically for string fields. It seems to me that there is no need to add it. This log-file is a real-life example from the stdout of the LA agent.

File: Parsers/Corelight/Corelight (hunk 4 above)
Comment (averbn) [reply Parsers/Corelight/Corelight:1]: Ok, I`ll rename it

File: Parsers/Corelight/Corelight (hunk 6 above)
Comment (averbn) [reply Parsers/Corelight/Corelight:14]: _s suffix will be added automatically for string fields. It seems to me that there is no need to add it. This log-file is a real-life example from the stdout of the LA agent.

File: Workbooks/WorkbooksMetadata.json [hunk 7]
```diff
@@ -1308,6 +1308,19 @@
     "title": "pfsense",
     "templateRelativePath": "pfsense.json",
     "subtitle": "",
-    "provider": "Azure Sentinel community"
+    "provider": "McAfee"
+  },
+  {
+    "workbookKey": "CorelightWorkbook",
+    "logoFileName": "corelight.svg",
```
[the remark this answers is not in the document]
Comment (v-jayakal) [reply Workbooks/WorkbooksMetadata.json:1315]: @averbn - Can you please provide us the logo mentioned in the PR (corelight.svg)?

**Review by shainw — COMMENTED** [review]
Approved from the Parser perspective

**Comment by v-admahe** [comment]
@averbn  Reverted the metadata json file changes as this going to be deployed as solution.

File: Workbooks/WorkbooksMetadata.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Workbooks/WorkbooksMetadata.json]: Sure

**Comment by v-admahe** [comment]
@averbn  Workbook queries are not appropriate. Could you please make the required changes in the queries. Please find the screenshot of the queries below in the zip file.
[CorelightQuery.zip](https://github.com/Azure/Azure-Sentinel/files/6170508/CorelightQuery.zip)

Thanks,
Aditi Maheshwari

**Comment by averbn** [comment]
> @averbn Workbook queries are not appropriate. Could you please make the required changes in the queries. Please find the screenshot of the queries below in the zip file.
> [CorelightQuery.zip](https://github.com/Azure/Azure-Sentinel/files/6170508/CorelightQuery.zip)
> 
> Thanks,
> Aditi Maheshwari

Fixed, pls check

**Review by v-jayakal — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Comment by v-jayakal** [comment]
@averbn - Logo has some invalid attributes like data-name, title. Updated the logo to follow the guideline and merged the content. Please check the logo and if you want to provide another version, please update it. As Corelight is to be deployed as solution, we will be moving the content to the solution folder and do the required like validations etc from solutions folder. Please let us know if you are okay with changes to the logo OR if you want to give another version. Thanks.

**Pull request merged**